test/fakeintake/server/parser.go

```diff
@@ -78,14 +73,11 @@ func (fi *PayloadParser) parse(payload api.Payload, route string) {
 		Encoding:  payload.Encoding,
 	}
 
-	if route == "/api/v2/logs" {
-		parsedPayload.Data = fi.getLogPayLoadJson(payload)
-	} else if route == "/api/v2/series" {
-		parsedPayload.Data = fi.getMetricPayLoadJson(payload)
-	} else if route == "/api/v1/check_run" {
-		parsedPayload.Data = fi.getCheckRunPayLoadJson(payload)
+	if payloadFunc, ok := fi.parserMap[route]; ok {
+		parsedPayload.Data, _ = payloadFunc(payload)
```

Comment:
Is there a reason to ignore the error?

Reply:
In general, it is better to avoid ignoring an error because a new error may be added to the function which invalidate your assumption.